ir/shared/src/main/scala/org/scalajs/ir/Trees.scala

```diff
@@ -966,7 +966,19 @@ object Trees {
      *  [[ReservedJSIdentifierNames]]).
      */
     def isValidJSGlobalRefName(name: String): Boolean =
-      isJSIdentifierName(name) && !ReservedJSIdentifierNames.contains(name)
+      (isJSIdentifierName(name) && !ReservedJSIdentifierNames.contains(name)) ||
+          name == FileLevelThis
+
+    /** The JavaScript value that is an alias to `this`
+     *  at the top-level of the generated file.
+     *
+     *  Backends are expected to:
+     *  - Generate `const fileLevelThis = this` at the top level of the JS file.
+     *  - Transform `JSGlobalRef("this")` to reference the `fileLevelThis` variable.
+     *
+     *  Otherwise, `this` might not refer to the global object, depending on the context.
```

Comment:
```suggestion
```
The IR has no business telling the backends *how* they should do things. ;) The IR describes the *semantics* (the "what") only.

For example, an IR interpreter would definitely not follow the above implementation.

Reply:
> The IR has no business telling the backends how they should do things.

Right... removed!